plugin/src/withPodfile.ts

```diff
@@ -20,41 +19,47 @@ export const withPodfile: ConfigPlugin<{
           : "use_expo_modules!";
 
       const appClipTarget = `
-        target '${targetName}' do
-          ${useExpoModules}
-          config = use_native_modules!
+target '${targetName}' do
+  ${useExpoModules}
 
-          use_frameworks! :linkage => podfile_properties['ios.useFrameworks'].to_sym if podfile_properties['ios.useFrameworks']
-          use_frameworks! :linkage => ENV['USE_FRAMEWORKS'].to_sym if ENV['USE_FRAMEWORKS']
+  if ENV['EXPO_USE_COMMUNITY_AUTOLINKING'] == '1'
+    config_command = ['node', '-e', "process.argv=['', '', 'config'];require('@react-native-community/cli').run()"];
+  else
+    config_command = [
+      'node',
+      '--no-warnings',
+      '--eval',
+      'require(require.resolve(\\'expo-modules-autolinking\\', { paths: [require.resolve(\\'expo/package.json\\')] }))(process.argv.slice(1))',
+      'react-native-config',
+      '--json',
+      '--platform',
+      'ios'
+    ]
+  end
+  config = use_native_modules!(config_command)
 
-          use_react_native!(
-            :path => config[:reactNativePath],
-            :hermes_enabled => podfile_properties['expo.jsEngine'] == nil || podfile_properties['expo.jsEngine'] == 'hermes',
-            # An absolute path to your application root.
-            :app_path => "#{Pod::Config.instance.installation_root}/..",
-            :privacy_file_aggregation_enabled => podfile_properties['apple.privacyManifestAggregationEnabled'] != 'false',
-          )
-        end
-      `;
+  use_frameworks! :linkage => podfile_properties['ios.useFrameworks'].to_sym if podfile_properties['ios.useFrameworks']
+  use_frameworks! :linkage => ENV['USE_FRAMEWORKS'].to_sym if ENV['USE_FRAMEWORKS']
 
-      /* podfileContent = podfileContent
-        .concat(`\n\n# >>> Inserted by react-native-app-clip`)
-        .concat(podfileInsert)
-        .concat(`\n\n# <<< Inserted by react-native-app-clip`); */
+  use_react_native!(
+    :path => config[:reactNativePath],
+    :hermes_enabled => podfile_properties['expo.jsEngine'] == nil || podfile_properties['expo.jsEngine'] == 'hermes',
+    # An absolute path to your application root.
+    :app_path => "#{Pod::Config.instance.installation_root}/..",
+    :privacy_file_aggregation_enabled => podfile_properties['apple.privacyManifestAggregationEnabled'] != 'false',
+  )
+end
+      `.trim();
 
-      podfileContent = mergeContents({
-        tag: "react-native-app-clip-2",
-        src: podfileContent,
-        newSrc: appClipTarget,
-        anchor: "Pod::UI.warn e",
-        offset: 3,
-        comment: "#",
-      }).contents;
+      // It's generally safer to use `mergeContents` to insert new content into the Podfile, but this new target should always be inserted at the end of the file.
+      podfileContent = podfileContent
+        .concat(`\n\n# >>> Inserted by react-native-app-clip\n`)
+        .concat(appClipTarget)
+        .concat(`\n\n# <<< Inserted by react-native-app-clip\n`);
```

Comment:
Why did you get rid of `mergeContents`?

Reply:
The previous `anchor` (`Pod::UI.warn e) was removed in Expo 52 and honestly was quite arbitrary. I believe it was chosen just because it happened to be the last identifiable line in the `Podfile`. Inserting at the end is much safer and resilient to future `Podfile` changes, especially when considering interactions with other plugins.